components/dashboard/dashboard.go

```diff
@@ -25,21 +25,20 @@ import (
 )
 
 var (
-	ComponentName   = "dashboard"
-	Path            = deploy.DefaultManifestPath + "/" + ComponentName + "/base"        // ODH
-	PathISV         = deploy.DefaultManifestPath + "/" + ComponentName + "/apps"        // ODH APPS
-	PathCRDs        = deploy.DefaultManifestPath + "/" + ComponentName + "/crd"         // ODH + RHOAI
-	PathConsoleLink = deploy.DefaultManifestPath + "/" + ComponentName + "/consolelink" // ODH consolelink
-
-	ComponentNameSupported   = "rhods-dashboard"
-	PathSupported            = deploy.DefaultManifestPath + "/" + ComponentName + "/overlays/rhoai"              // RHOAI
-	PathISVSM                = deploy.DefaultManifestPath + "/" + ComponentName + "/overlays/apps/apps-onprem"   // RHOAI APPS
-	PathISVAddOn             = deploy.DefaultManifestPath + "/" + ComponentName + "/overlays/apps/apps-addon"    // RHOAI APPS
-	PathConsoleLinkSupported = deploy.DefaultManifestPath + "/" + ComponentName + "/overlays/consolelink"        // RHOAI
-	PathODHDashboardConfig   = deploy.DefaultManifestPath + "/" + ComponentName + "/overlays/odhdashboardconfig" // RHOAI odhdashboardconfig
-
-	NameConsoleLink      = "console"
-	NamespaceConsoleLink = "openshift-console"
+	ConsoleLinkName = "console"
+	ConsoleLinkNS   = "openshift-console"
+
+	ComponentNameUpstream   = "dashboard"
+	PathUpstream            = deploy.DefaultManifestPath + "/" + ComponentNameUpstream + "/odh"
+	ConsoleLinkPathUpstream = PathUpstream + "/consolelink"
+
+	ComponentNameDownstream   = "rhods-dashboard"
+	PathDownstream            = deploy.DefaultManifestPath + "/" + ComponentNameUpstream + "/rhoai"
+	PathSelfDownstream        = PathDownstream + "/onprem"
+	PathManagedDownstream     = PathDownstream + "/addon"
+	ConsoleLinkPathDownstream = PathDownstream + "/shared/consolelink"
+	PathSupported             = PathDownstream + "/shared/base"               // params.env
+	PathODHDashboardConfig    = PathDownstream + "/shared/odhdashboardconfig" // RHOAI odhdashboardconfig
```

Comment:
Do we need to know internals of our paths?

I'm hoping all you need to know is:
- ODH is at `manifests/odh`
- RHOAI OnPrem is at `manifests/rhaoi/onprem`
- RHOAI AddOn is at `manifests/rhoai/addon`

Effectively anything else should be done through params or some "agnostic" way... you having knowledge of our other paths feels like more break points if we need to move or update things.

Reply:
these are 3 special cases:
- consolelink , it requires to replace "section title" and "dashboard-url". "section title" probably you can make it hardcoded by kustomization patch, but the "dashboard-url" will be dynamically generated with domain and application namespace. 
- odhdashboardconfig,  has the "admin_groups", similar to "section title" it could also be patched with kustomization if you want to do.
- the last is the params.env, Operator will need explicit path to get that file. 

apart from these, others are following this pattern:
- ODH is at manifests/odh
- RHOAI OnPrem is at manifests/rhaoi/onprem
- RHOAI AddOn is at manifests/rhoai/addon

Reply:
Unfortunately this is not good enough lol -- we need to talk about breaking this model where the Operator tracking locations of files. I don't mind "relative paths" -- like the root of the deployment or something.

We absolutely need a mechanism that we can apply variables to the manifests in a way that allows the power of the usage to live in the component. If the Operator manages everything, there is an invisible brittle link there. 

Lets take this opportunity and strive to get to a better place instead of carrying forward this model.

Reply:
I would prefer these logic to be moved into component (e.g dashboard) itself rather than a lot need to be done by Operator.

But i would say this is not in the scope of the original ticket.  if we want to refactor manifests struct,unless expand that ticket? 
Should we do a follow-up with another set of changes for the new logic? I would expect the ticket to be mostly driven by dashboard, then Operator only do a cleanup to remove these calls once such function is available in dashboard

Reply:
For posterity, we have taken this conversation to a meeting on Monday with Vaishnavi. TBD if we address this "now" or on another pass with other issues.

Reply:
For posterity (again), we had our meeting and @zdtsw will update her PR to not include internal paths to RHOAI and update it to inject variables into the root `params.env` for each deployment.

Reply:
We can mark this thread resolved if you want @zdtsw